### PR TITLE
reworked PR containing all commits

### PR DESCRIPTION
**Pull request by fabrice-etanchaud**
Hello, This PR contains : 

- a full implementation of adapters.sql and catalog.sql
- setup.py updated to match latest 0.17.2 release version

Now, documentation gets generated, and description ends up in db's comments !

Concerning custom schema tests, I already pushed a adapter like version, but we will have to wait for the next release in order to override default ones. For now, one has to copy the schema_test.sql file into the main project, and remove the "oracle__" prefix of the macro names.

I hope you will be able to test and give me feedback soon !

Thank you for initiating this work, as we are still a few to be stucked on oracle.

### CODE REVIEW

**Comment by vitoravancini** [comment]
Hello @fabrice-etanchaud, sorry for taking this long to answer here, its been a long couple of weeks.

Ill take a look tonight and try to test it.

Thank you for commiting! We will probably be stucked with Oracle for a long time too! haha

File: dbt/include/oracle/macros/materializations/seed/seed.sql [hunk 1]
```diff
@@ -7,17 +8,20 @@
...
+
         {% set sql %}
-            insert into {{ this.render() }} ({{ cols_sql }})
+            insert all
             {% for row in chunk -%}
-                select
+              into {{ this.render() }} ({{ cols_sql }}) values(
                 {%- for column in agate_table.column_names -%}
-                    {{" '" + row[loop.index - 1] + "' " }}
+                    :p{{ loop.index }}
```
Comment (vitoravancini) [inline dbt/include/oracle/macros/materializations/seed/seed.sql:20]: I have the same question I had in your other PR. Is this ':p' a jinja feature? Is it a typo?

Comment (fabrice-etanchaud) [reply dbt/include/oracle/macros/materializations/seed/seed.sql:20]: Hi @vitoravancini , no, this is how you define binded variables in a parametrized statement (I did not know that a few weeks ago)

[](https://cx-oracle.readthedocs.io/en/latest/user_guide/bind.html)

**Comment by dpavancini** [comment]
@vitoravancini can you take a look into that?

File: dbt/include/oracle/macros/materializations/seed/seed.sql (hunk 1 above)
Comment (vitoravancini) [reply dbt/include/oracle/macros/materializations/seed/seed.sql:20]: Ah allright!

**Comment by vitoravancini** [comment]
@fabrice-etanchaud Looks good, going to merge it. Thank you very much for all your contribution!

Gonna bump to v0.2.0 and publish at pypi.

**Pull request merged**